// @adminjs/design-system/build/molecules/select/select.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import noop from 'lodash/noop.js';
import React, { lazy } from 'react';
import { cssClass, filterStyles, selectStyles } from "../../utils/index.js";
import useSelectTheme from "./select-theme.js";
const ReactSelect = /*#__PURE__*/lazy(() => import(function transformExtension(filepath, extMapping) {
  if (!filepath.startsWith('./') && !filepath.startsWith('../')) {
    // Package import
    return filepath;
  }
  const idx = filepath.lastIndexOf('.');
  if (idx === -1 || filepath.includes('/', idx)) {
    // No extension
    const newExt = extMapping[''];
    if (newExt) {
      return filepath + newExt;
    }
    return filepath;
  }
  for (let [origExt, newExt] of Object.entries(extMapping).sort((a, b) => b[0].length - a[0].length)) {
    if (filepath.endsWith(origExt)) {
      return filepath.slice(0, -origExt.length) + newExt;
    }
  }
  return filepath;
}('react-select', {
  ".jsx": ".js"
})));
const SelectComponent = ReactSelect.default || ReactSelect;
export const Select = props => {
  const {
    value,
    onChange,
    variant,
    isMulti,
    ...selectProps
  } = props;
  const {
    theme,
    selectTheme
  } = useSelectTheme();
  const styles = variant === 'filter' ? filterStyles(theme) : selectStyles(theme);
  const handleChange = selected => {
    if (typeof onChange === 'function') onChange(selected);
  };
  return /*#__PURE__*/React.createElement(SelectComponent, _extends({
    className: cssClass('Select'),
    value: value,
    theme: selectTheme,
    styles: styles,
    onChange: handleChange,
    isClearable: true,
    isMulti: isMulti
  }, selectProps));
};
Select.defaultProps = {
  variant: 'default',
  onChange: noop
};
Select.displayName = 'Select';
export default Select;
